HW10/script.js

```diff
@@ -0,0 +1,169 @@
+/** Завдання створити класи:
+ *    фігура:
+ *      координати:
+ *      х, у
+ *
+ *    багатокутник -> фігура:
+ *      вершини[координати],
+ *      периметр()
+ *
+ *    прямокутник -> багатокутник:
+ *      довжина,
+ *      висота,
+ *      площа(),
+ *      периметр()
+ *
+ *    квадрат -> прямокутник:
+ *      довжинаСторони,
+ *      площа(),
+ *      периметр()
+ *
+ *    круг -> фігура:
+ *      радіус,
+ *      площа(),
+ *      периметр()
+ */
+document.addEventListener('DOMContentLoaded', function () {
+    /**
+     * @param {Point} center
+     */
+    class Shape {
+        constructor(center) {
+            this.center = center;
+        }
+    }
+
+    /**
+     * @param {Point} center
+     * @param {points[]} points
+     */
+    class Polygon {
+        constructor(center, points) {
+            this._center = center;
+            this._points = points;
+        }
+        /**
+         *
+         * @return {number}
+         */
+        perimeter() {
+            let sum = 0;
+            if (this._points.length >= 4) {
+                for (let i = 0; i < this._points.length - 1; i++) {
+                    sum += this._points[i].getDistance(this._points[i + 1]);
+                }
+            }
+            return sum;
+        }
+    }
+
+    class Rectangle {
+        constructor(center, width, height) {
+            this._center = center;
+            if (this._points.length === 4) {
+                this._width = this._points[0].getDistance(this._points[1]);
+                this._height = this._points[1].getDistance(this._points[3]);
+            }
+            else if (this._points.length !== 4) {
```

Comment:
```suggestion
            if (this._points.length !== 4) {
```
и все что перед этим ифом с эррором перемести за ним. И тот иф что выше не нужен

Reply:
погодика, а откуда у тебя вообще `_points` взялись? они же `undefined`